### PR TITLE
Fixups to allow running out of subdirectory

### PR DESCRIPTION
**Pull request by r0p0s3c**
Currently, there are a number of issues with running rocket.chat out of a subdirectory of the webroot (e.g. <host>/chat). These seem to be primarily around resources/assets that are referenced, such as fonts, images and so on. I have also opened a number of issues on modules being used (e.g. sandstorm-io/meteor-accounts-sandstorm#17, mizzao/meteor-timesync#36) and will issue another PR once those are wrapped up. In the meantime this PR should fix the assumption in rocket.chat itself, and help to close a number of issues (e.g. #1696, #335, #2401)

### CODE REVIEW

**Comment by sampaiodiego** [comment]
what happens when your first hit is in a URL like this https://demo.rocket.chat/channel/general ?
are the assets loaded correctly?

**Comment by r0p0s3c** [comment]
Hi,

all assets (fonts, sounds, avatar images, etc.) do not load correctly (i.e. 404s) without those updates for me. The issue is fairly easy to replicate - setting ROOT_URL to http://localhost:3000/chat should do it.

Thanks

**Comment by sampaiodiego** [comment]
I'm sorry.. I mean, **with** your fix, what happens when your first hit is in a URL like this https://demo.rocket.chat/channel/general ?

**Comment by r0p0s3c** [comment]
Hi,

on the instance I am testing, where ROOT_URL is set to <host>/chat/, going to <host>/chat/channel/general for the first time returns the login screen. Are you seeing a regression for this?

**Comment by engelgabriel** [comment]
Hi @r0p0s3c we will merge this into develop tomorrow to give enough time for testers, can you just please resolve the conflicts?

File: packages/rocketchat-lib/client/lib/settings.coffee [hunk 1]
```diff
@@ -41,12 +41,11 @@ Meteor.startup ->
...
 				return c.stop()
 
-			siteUrl = siteUrl.replace /\/$/, ''
-			if siteUrl isnt location.origin
+			if siteUrl isnt Meteor.absoluteUrl()
 				swal
 					type: 'warning'
 					title: t('Warning')
-					text: t("The_setting_s_is_configured_to_s_and_you_are_accessing_from_s", t('Site_Url'), siteUrl, location.origin) + '<br/><br/>' + t("Do_you_want_to_change_to_s_question", location.origin)
+					text: t("The_setting_s_is_configured_to_s_and_you_are_accessing_from_s", t('Site_Url'), siteUrl, Meteor.absoluteUrl()) + '<br/><br/>' + t("Do_you_want_to_change_to_s_question", Meteor.absoluteUrl())
```
Comment (rodrigok) [inline packages/rocketchat-lib/client/lib/settings.coffee]: siteUrl and Meteor.absoluteUrl() will always be the same.
This verification exists to warn users when they access from an URL different from the siteUrl

**Comment by rodrigok** [comment]
Closed in favor of https://github.com/RocketChat/Rocket.Chat/pull/2624

**Comment by r0p0s3c** [comment]
Thanks, was not able to get around to commenting on the comments in time

**Pull request closed without merging**